pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -36,13 +38,47 @@ const (
 )
 
 type StorageInitializerConfig struct {
-	Image string `json:"image"`
+	Image         string `json:"image"`
+	CpuRequest    string `json:"cpuRequest"`
+	CpuLimit      string `json:"cpuLimit"`
+	MemoryRequest string `json:"memoryRequest"`
+	MemoryLimit   string `json:"memoryLimit"`
 }
+
 type StorageInitializerInjector struct {
 	credentialBuilder *credentials.CredentialBuilder
 	config            *StorageInitializerConfig
 }
 
+func getStorageInitializerConfigs(configMap *v1.ConfigMap) (*StorageInitializerConfig, error) {
+	storageInitializerConfig := &StorageInitializerConfig{}
+	if initializerConfig, ok := configMap.Data[StorageInitializerConfigMapKeyName]; ok {
+		err := json.Unmarshal([]byte(initializerConfig), &storageInitializerConfig)
+		if err != nil {
+			panic(fmt.Errorf("Unable to unmarshall %v json string due to %v ", StorageInitializerConfigMapKeyName, err))
+		}
+	}
+	//Ensure that we set proper values for CPU/Memory Limit/Request
+	_, err := resource.ParseQuantity(storageInitializerConfig.MemoryRequest)
```

Comment:
This series of functions looks like a perfect opportunity for a for loop.

for _, key := range {
   storageInitializerConfig.MemoryRequest,
   storageInitializerConfig.MemoryLimit,
   ...
} {
 parse quantity
 return if error
}

Reply:
Good point!